Update release versions to 0.6

grantr opened the pull request:
Now that 0.6 has been released, we can update all the install links to `v0.6.0`.

sebgoa commented on docs/install/Knative-with-Docker-for-Mac.md:
````diff
@@ -37,7 +37,7 @@ Knative depends on Istio. Run the following to install Istio. (This changes
 
 ```shell
-curl -L https://raw.githubusercontent.com/knative/serving/v0.5.2/third_party/istio-1.0.7/istio.yaml \
+curl -L https://raw.githubusercontent.com/knative/serving/v0.6.0/third_party/istio-1.0.7/istio.yaml \
````
Still Istio 1.0.7 ?

samodell replied:
I think @grantr  made this PR before #1314 was merged in, which removed the references to our provided Istio files in 0.6. The new instructions from @ZhiminXiang  now point users to download Istio directly from their website, and they instruct folks to download Istio 1.1.3: https://github.com/knative/docs/pull/1314/files#diff-1116bcf887cf6b5e21e5e4f96fa026b1

samodell replied:
Ah, actually, I was going to save updating this file for later, since the Istio install here is doing something different, and I'm not sure it's possible to do this LoadBalancer/NodePort switch when installing Istio through `helm`?

We shouldn't update the reference here to 0.6, because we did not release an Istio.yaml for 0.6. We should leave in the reference to 0.5.2 for now, until these instructions can be updated. I created an issue to track the work -- #1319

grantr replied:
@samodell would you like me to revert **all** changes to the Minikube, Minishift, Openshift, and Docker for Mac files, or just the istio URLs?

ZhiminXiang replied:
> Ah, actually, I was going to save updating this file for later, since the Istio install here is doing something different, and I'm not sure it's possible to do this LoadBalancer/NodePort switch when installing Istio through `helm`?

@samodell It is possible. you can configure helm flag `gateways.istio-ingressgateway.type` to choose the gateway service type. See https://istio.io/docs/reference/config/installation-options/#gateways-options

sebgoa commented on docs/install/Knative-with-Minikube.md:
````diff
@@ -66,8 +66,8 @@ Knative depends on Istio. Run the following to install Istio. (We are changing
 ```shell
-kubectl apply --filename https://raw.githubusercontent.com/knative/serving/v0.5.2/third_party/istio-1.0.7/istio-crds.yaml &&
-curl -L https://raw.githubusercontent.com/knative/serving/v0.5.2/third_party/istio-1.0.7/istio.yaml \
+kubectl apply --filename https://raw.githubusercontent.com/knative/serving/v0.6.0/third_party/istio-1.0.7/istio-crds.yaml &&
````
same here, 1.0.7 ?

samodell commented:
Hey @grantr  -- if you're still up for it, it would be great to revert all the changes to the files. I think that's better than mismatching V0.6 with outdated Istio instructions.

grantr commented:
> if you're still up for it, it would be great to revert all the changes to the minkube/minishift/openshift/etc. files.

No problem! I rebased on top of the latest master and reverted the changes to the Minikube, Minishift, OpenShift, and Docker for Mac install instructions.

samodell commented:
/lgtm
/approve
